loki/transformations/single_column/tests/test_scc.py

```diff
@@ -1407,3 +1407,92 @@ def test_scc_resolve_vector_dimension(frontend, horizontal):
 
     assert 'maxval' == assigns[1].rhs.name.lower()
     assert 'start:end' in assigns[1].rhs.parameters[0].dimensions
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_scc_annotate_driver_loop(frontend, tmp_path):
+    """
+    Test for issue #246 to ensure the correct loop in the loop nest
+    is identified and annotated as driver loop
+    """
+    fcode = """
+module mod
+    implicit none
+    integer, parameter :: ngpblk = 2
+    integer, parameter :: klon = 10
+    integer, parameter :: klev = 5
+    contains
+    subroutine wrapper(arr)
+        integer, parameter :: niter = 10
+        integer, intent(in) :: arr(klon, klev, ngpblk)
+        integer :: jblk, jiter
+
+        ! Begin relevant part.
+        DO jiter=1,niter
+            DO jblk=1,ngpblk
+                CALL kernel(klon, klev, 1, 1, arr(:, :, jblk))
+            END DO
+        END DO
+        ! End relevant part.
+
+    end subroutine wrapper
+
+    subroutine kernel(klon, klev, jlev_lower, jlon_lower, arr)
+        integer, intent(in) :: klon
+        integer, intent(in) :: klev
+        integer, intent(in) :: jlev_lower, jlon_lower
+        integer, intent(inout) :: arr(klon, klev)
+        integer :: jlon, jlev
+
+        DO jlev=jlev_lower,klev
+            DO jlon=jlon_lower,klon
+                arr(jlon, jlev) = jlon + jlev
+            END DO
+        END DO
+
+    end subroutine kernel
+
+end module mod
+    """.strip()
+
+    source = Sourcefile.from_source(fcode, frontend=frontend, xmods=(tmp_path,))
+
+    horizontal = Dimension('horizontal', index='JLON', bounds=['jlon_lower', 'klon'], size='KLON')
+    block_dim = Dimension('block_dim', index='JBLK', size='NGPBLK')
+
+    # Test OpenACC annotations on non-hoisted version
+    scc_transform = (SCCDevectorTransformation(horizontal=horizontal),)
+    scc_transform += (SCCDemoteTransformation(horizontal=horizontal),)
+    scc_transform += (SCCRevectorTransformation(horizontal=horizontal),)
```

Comment:
I appreciate this was written a while ago, but I think we can now use the respective `SCCVectorPipeline` (or equivalent) to make this a single invocation line.